ding_unilogin/src/Controller/InstitutionController.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace Drupal\ding_unilogin\Controller;
+
+use Drupal\ding_unilogin\Exception\HttpBadRequestException;
+use Drupal\ding_unilogin\Exception\HttpNotFoundException;
+use Drupal\ding_unilogin\Exception\HttpUnauthorizedException;
+
+/**
+ * Institution controller.
+ */
+class InstitutionController {
+
+  /**
+   * Handle request.
+   */
+  public function handle(array $args) {
+    $authorization = $this->getRequestHeader('authorization');
+    if (!preg_match('/^(bearer|token) (?P<token>.+)$/i', $authorization, $matches)) {
+      throw new HttpUnauthorizedException();
+    }
+    $token = $matches['token'];
+
+    if (empty($args)) {
+      $method = $_SERVER['REQUEST_METHOD'];
+
+      if ('POST' === $method) {
+        if (variable_get('ding_unilogin_api_token_write') !== $token) {
+          throw new HttpUnauthorizedException();
+        }
+
+        return $this->update();
+      }
+
+      if (variable_get('ding_unilogin_api_token_read') !== $token) {
```

Comment:
So PUT, PATCH or DELETE request would get the read access (if valid token)? A `not supported` http response might be more correct.

Reply:
Good point!